app/models/labware_creators/common_file_handling/csv_file/row_for_tube_rack_with_rack_barcode.rb

```diff
@@ -24,6 +25,10 @@ def initialize_context_specific_fields
         @tube_position = (@row_data[1] || '').strip.upcase
         @tube_barcode = (@row_data[2] || '').strip.upcase
       end
+
+      def expected_number_of_columns
+        EXPECTED_NUMBER_OF_COLUMNS
+      end
```

Comment:
I don't like this because you're defining the exact same method in the subclass (RowForTubeRackWithRackBarcode) and the superclass (RowForTubeRack). So, it looks like you could just get rid of this method in the subclass and it would inherit it instead. However, if you do this, it looks like it would behave incorrectly and always read the `EXPECTED_NUMBER_OF_COLUMNS` constant from the superclass (https://www.ruby-forum.com/t/class-inheritance-and-class-constants/210683)

So personally I would just get rid of the constants and put the number 2 or 3 directly in the methods.

Reply:
Ok will change it.
It does do that inheritance thing with the constants, found that out the hard way before. very confusing.